.github/workflows/test-doc-build.yml

```diff
@@ -0,0 +1,20 @@
+name: Test the documentation build
+
+on: [pull_request, workflow_dispatch]
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.11'
+        cache: 'pip'
+        cache-dependency-path: '.github/workflows/doc-requirements.txt'
+    - run: python -m pip install -r .github/workflows/doc-requirements.txt
+    - run: python -m pip install .
+    - run: |
+        python -m sphinx.ext.apidoc --separate --no-toc -o docs/source/api -t docs/source/api/templates envisage */tests
+        python -m sphinx -b html docs/source docs/build
```

Comment:
Done in 31b2c8cd716eb0286843fddce04a1f7d533e5336

I've tweaked the Sphinx build command to move the doctrees (which we don't need) out of the build directory, and make the build directory `build/html` rather than plain old `build`.

Let's see if the built artifact is useful.
Potentially, this could be the first step in an automated build-docs-at-release-time workflow.

Reply:
Build artifact seems to be working. I'd be a bit happier if the `.zip` file gave something with an `html` top-level directory. I've just pushed a change that _might_ just make that happen.